contracts/upgradeable_contracts/ChaiConnector.sol

```diff
@@ -252,25 +256,28 @@ contract ChaiConnector is Ownable, ERC20Bridge {
     */
     function _convertChaiToDai(uint256 amount) internal {
         uint256 invested = investedAmountInDai();
+        uint256 newInvestedAmountInDai;
         uint256 initialDaiBalance = daiBalance();
         if (amount >= invested) {
             // onExecuteMessage can call a convert operation with argument greater than the current invested amount,
             // in this case bridge should withdraw all invested funds
             chaiToken().draw(address(this), invested);
-            setInvestedAmointInDai(0);
+
+            newInvestedAmountInDai = 0;
 
             // Make sure all invested tokens were withdrawn
             require(daiBalance() - initialDaiBalance >= invested);
         } else if (amount > 0) {
             chaiToken().draw(address(this), amount);
             uint256 redeemed = daiBalance() - initialDaiBalance;
 
+            newInvestedAmountInDai = redeemed < invested ? invested - redeemed : 0;
             // Make sure that at least requested amount was withdrawn
             require(redeemed >= amount);
-
-            setInvestedAmointInDai(redeemed < invested ? invested - redeemed : 0);
         }
 
-        require(dsrBalance() >= investedAmountInDai());
+        setInvestedAmountInDai(newInvestedAmountInDai);
```

Comment:
If the `amount` is `0` and `invested` is not zero, this call will reset the `InvestedAmountInDai`.

Reply:
Added condition on `amount > 0`, please take a look.

Reply:
I'd propose to rewrite the `_convertChaiToDai` internal function as follows:

```solidity
function _convertChaiToDai(uint256 amount) internal {
    if (amount == 0) return;

    uint256 invested = investedAmountInDai();
    uint256 initialDaiBalance = daiBalance();

    // onExecuteMessage can call a convert operation with argument greater than the current invested amount,
    // in this case bridge should withdraw all invested funds
    uint256 withdrawal = amount >= invested ? invested : amount;

    chaiToken().draw(address(this), withdrawal);
    uint256 redeemed = daiBalance() - initialDaiBalance;

    // Make sure that at least withdrawal amount was withdrawn
    require(redeemed >= withdrawal);

    uint256 newInvested = invested > redeemed ? invested - redeemed : 0;
    setInvestedAmountInDai(newInvested);

    require(dsrBalance() >= newInvested);
}
```

Also, let's add unit tests for this function which would cover the cases for `amount = 0`, `amount >= investedAmountInDai()`, and `amount < investedAmountInDai() && amount > 0`.